### PR TITLE
[fix](Nereids): fix AssertNumRows StatsCalculator

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

AssertNumRows should correct rowcount of child plan

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 1]
```diff
@@ -556,11 +558,16 @@ public Statistics visitPhysicalGenerate(PhysicalGenerate<? extends Plan> generat
...
         Statistics statistics = groupExpression.childStatistics(0);
-        statistics.withRowCountAndEnforceValid(Math.min(1, statistics.getRowCount()));
-        statistics = new StatisticsBuilder(statistics).setWidthInJoinCluster(1).build();
-        return statistics;
+        if (assertNumRowsElement.getAssertion() == Assertion.NE
+                || assertNumRowsElement.assertNumber((long) Math.ceil(statistics.getRowCount()))) {
+            return statistics;
+        } else {
+            Statistics newStatistics = statistics.withRowCountAndEnforceValid(
+                    Math.min(assertNumRowsElement.getDesiredNumOfRows(), statistics.getRowCount()));
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:568]: if assetion is GT

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Review by keanji-x — APPROVED** [review]

**Comment by jackwener** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**